File: Final Project/CarRentApi/BOL/UserModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace BOL
{
    public class UserModel
    {
        [Required, MinLength(3), MaxLength(100)]
        public string FullName { get; set; }

        [Required]
        [IdValidator]
        public string IdCard { get; set; }

        [Required, MinLength(5), MaxLength(20)]
        public string UserName { get; set; }

        
        public DateTime? DoB { get; set; }

        [Required]
        public bool IsMale { get; set; }

        [Required]
        public string EmailAddress { get; set; }

        [Required]
        public string Passwd { get; set; }

        [Required]
        public UserRoleModel Role { get; set; }

        
        public string Img { get; set; }



    }
}

File: Final Project/CarRentApp/src/app/shared/models/user.model.ts
import { UserRole } from "./user-role.model";

export interface User{
     FullName:string,
     IdCard:string,
     UserName:string,
     DoB?: Date,
     IsMale: boolean,
     EmailAddress:string,
     Passwd: string,
     Role:UserRole,
     Img?:string
     
}